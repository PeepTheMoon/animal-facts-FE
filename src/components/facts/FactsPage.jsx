import React, { useEffect } from 'react';
import Facts from './Facts';
import { useDispatch, useSelector } from 'react-redux';
import { setFacts, setFactsLoading } from '../../actions/factsActions';
import { getFactsLoading } from '../selectors/factsSelectors';
import { fetchFacts } from '../../../services/fetchFacts';

const FactsPage = () => {
  const dispatch = useDispatch();
  const loading = useSelector(getFactsLoading);

  useEffect(() => {
    dispatch(setFactsLoading());
    fetchFacts()
      .then(facts => {
        dispatch(setFacts(facts));
      });
  }, []);

  if(loading) return <h1>Loading Facts</h1>;

  return (
    <Facts />
  );
};

export default FactsPage;
