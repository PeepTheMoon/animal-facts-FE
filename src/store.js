import { createStore } from 'redux';
import reducer, { initialState } from './components/App/reducers/AppReducer';

const store = createStore(
  reducer, 
  initialState,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

export default store;
