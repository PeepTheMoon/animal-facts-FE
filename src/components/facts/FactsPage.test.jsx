import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import store from '../../store';
import FactsPage from './FactsPage';

jest.mock('../../../services/fetchFacts.js', () => ({
  fetchFacts() {
    return Promise.resolve([
      { image: 'image.png', caption: 'wild fact 1' },
      { image: 'image2.png', caption: 'wild fact 2' }
    ]);
  }
}));

describe('FactsPage component', () => {
  it('renders FactsPage', () => {
    const wrapper = mount(
      <Provider store={store}>
        <FactsPage />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
