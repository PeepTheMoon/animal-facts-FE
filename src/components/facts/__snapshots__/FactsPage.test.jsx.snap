// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FactsPage component renders FactsPage 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <FactsPage>
    <h1>
      Loading Facts
    </h1>
  </FactsPage>
</Provider>
`;
